### PR TITLE
Support max pooling while keeping the batch dimension

### PR DESCRIPTION
**Pull request by ChristopherBrix**
With this PR, it becomes possible to perform a max-/min pooling operation only across the time-dimension, while keeping the batch dimension untouched.

For `tf.reduce_max` and `tf.reduce_min`, we can replace all masked values with the minimal/maximal possible value. This way, they won't be selected during max/min operation.

### CODE REVIEW

**Comment by albertz** [comment]
I merged it in. Note though that there is a small (probably minor) problem: E.g. for mode="min", if your input has only values of `dtype.max` (i.e. not outside of the sequence), then this value will be the output of this layer. Which is correct. The problem is the gradient: The error signal to this min value will get forwarded normally to the `argmin` entry of the input. If there are multiple min values, it will be distributed across them (see `_MinOrMaxGrad` in TF). And this factor will be slightly wrong now. But as I said, probably this is really minor and does not matter.
